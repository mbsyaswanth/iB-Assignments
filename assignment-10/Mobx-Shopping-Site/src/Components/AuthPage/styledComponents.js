import styled from "styled-components";

export const AuthContainer = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  padding-top: 20vh;
  align-items: center;
  flex-direction: column;
`;

export const AuthHeading = styled.h2`
  font-weight: 700;
`;

export const AuthForm = styled.form`
  display: flex;
  width: 400px;
  padding: 30px;
  flex-direction: column;
  align-content: center;
  box-shadow: 0px 0px 2px 0px grey;
  border-radius: 2px;
  justify-content: space-evenly;
`;

export const InputContainer = styled.div`
  margin: 2px;
`;

export const FormText = styled.div`
  font-weight: 400;
  align-self: flex-start;
  margin-bottom: 10px;
`;

export const FormInput = styled.input`
  width: 100%;
  padding: 5px;
  margin-bottom: 20px;
  box-sizing: border-box;
`;
export const FormSubmit = styled.button`
  padding: 5px;
  margin-bottom: 10px;
`;

export const FormMsg = styled.div`
  margin-bottom: 10px;
  color: red;
  height: 1rem;
  padding: 3px;
  text-align: center;
`;

export const SignUpText = styled.div`
  padding: 5px;
  margin-top: 10px;
`;
