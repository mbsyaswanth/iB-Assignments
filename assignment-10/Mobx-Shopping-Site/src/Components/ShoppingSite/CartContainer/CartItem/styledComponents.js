import styled from "styled-components";

export const StyledCartItem = styled.div`
  display: flex;
  position: relative;
  padding-top: 10px;
  padding-bottom: 5px;
  box-shadow: 0px -3px 4px 0px black;
  margin-bottom: 10px;
`;

export const CartItemImage = styled.img`
  height: 100px;
`;

export const CartItemInfoWrapper = styled.div`
  flex-basis: 100%;
  align-self: center;
`;

export const CrossMark = styled.span`
  position: absolute;
  right: 5px;
  top: 5px;
  color: black;
  font-size: 1.5rem;
  font-weight: 800;
  padding-right: 5px;
  cursor: pointer;
  z-index: 1024;
`;

export const CartItemInfo = styled.div`
  padding: 10px;
  display: flex;
  justify-content: space-between;
  ${CrossMark}:hover & {
    text-decoration: line-through;
    font-weight: 800;
  }
`;

export const ItemName = styled.div`
  color: white;
  font-size: 1.3rem;
`;

export const ItemPrice = styled.div`
  display: flex;
  color: goldenrod;
  align-items: center;
`;

export const Grey = styled.div`
  color: grey;
`;
