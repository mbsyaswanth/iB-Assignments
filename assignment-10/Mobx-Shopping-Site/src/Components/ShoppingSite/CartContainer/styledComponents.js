import styled from "styled-components";

export const StyledCartContainer = styled.div`
  position: fixed;
  z-index: 999;
  display: flex;
  right: ${props => (props.isClicked ? "0" : "-40vw")};
  transition: 0.5s;
  height: 100vh;
`;

export const CartToggle = styled.div`
  position: relative;
  height: 60px;
  width: 60px;
  background-color: rgb(8, 7, 7);
  cursor: pointer;
  color: white;
  font-size: 42px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const CartCount = styled.span`
  display: ${props => (props.show ? "inline-block" : "none")};
  position: absolute;
  bottom: 5px;
  right: 5px;
  background-color: goldenrod;
  font-size: 12px;
  padding: 2px;
  border-radius: 50%;
`;

export const CartItemsContainer = styled.div`
  width: 40vw;
  background-color: rgb(8, 7, 7);
`;

export const InsideCartCount = styled(CartCount)`
  display: inline-block;
`;

export const CartItemsInnerWrap = styled.div`
  height: 77%;
  padding: 25px;
`;

export const CartIconInCart = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 15px;
`;

export const CartIcon = styled.img`
  height: 60px;
  width: 60px;
`;

export const InCartIcon = styled.img`
  height: 100px;
  width: 120px;
`;

export const CartText = styled.div`
  font-size: 20px;
  color: white;
`;

export const CartSubtotal = styled.div`
  height: 23%;
  padding: 25px;
  font-size: 16px;
  box-shadow: 0px -5px 20px 1px black;
`;

export const SubTotal = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const SubTotalText = styled.span`
  color: grey;
`;

export const SubTotalCount = styled.span`
  color: goldenrod;
`;

export const Checkout = styled.div`
  margin-top: 60px;
  color: white;
  background-color: black;
  padding: 15px;
  text-align: center;
  cursor: pointer;
`;
