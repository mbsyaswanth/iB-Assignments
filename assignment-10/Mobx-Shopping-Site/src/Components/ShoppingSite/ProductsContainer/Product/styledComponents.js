import styled from "styled-components";

export const ProductContainer = styled.div`
  position: relative;
  padding: 10px;
  margin: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 333px;
  background-color: white;
  transition: 0.2s;
  :hover {
    box-shadow: 0px 0px 2px 1px grey;
  }
`;

export const FreeShipping = styled.div`
  display: ${props => (props.isFreeShipping ? "block" : "none")};
  position: absolute;
  top: 10px;
  right: 10px;
  color: white;
  padding: 2px;
  background-color: black;
`;

export const ProductInfo = styled.div`
  display: flex;
  flex-direction: column;
  height: 160px;
  justify-content: space-evenly;
  align-items: center;
  margin-top: 10px;
`;

export const ProductTitle = styled.div`
  font-size: 1.2rem;
  height: 2.4rem;
`;

export const Line = styled.div`
  width: 30px;
  height: 2px;
  background-color: goldenrod;
`;

export const ProductPrice = styled.div`
  font-weight: 600;
  font-size: 1.6rem;
`;

export const ProductInstallments = styled.div`
  color: grey;
  font-weight: 600;
`;

export const ProductCartBtn = styled.button`
  width: 333px;
  padding: 10px;
  font-size: 1.1rem;
  color: white;
  background-color: black;
  font-weight: 500;
  border-style: none;
  cursor: pointer;
  transition: 0.2s;
  :focus {
    outline: none;
  }
  :active {
    border-style: none;
  }
  ${ProductContainer}:hover & {
    background-color: goldenrod;
  }
`;

export const Select = styled.select``;

export const Option = styled.option``;
