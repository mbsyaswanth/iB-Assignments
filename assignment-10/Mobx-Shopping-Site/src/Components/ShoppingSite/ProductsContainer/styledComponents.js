import styled from "styled-components";

export const ProductsContainerStyle = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
`;
