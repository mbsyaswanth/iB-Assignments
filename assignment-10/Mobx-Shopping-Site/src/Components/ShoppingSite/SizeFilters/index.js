import React, { Component } from "react";

import { observer } from "mobx-react";
import { action, computed, observable } from "mobx";

@observer
class SizeFilters extends Component {
  handleClick = event => {
    this.props.onFilterClick(event.target.value);
  };

  render() {
    return (
      <div className="size-filters-container">
        <div>
          <div className="size-filter-row1">
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XS")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XS"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("S")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="S"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("M")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="M"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("ML")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="ML"
            />
          </div>
          <div>
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("L")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="L"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XL")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XL"
            />
            <input
              type="button"
              onClick={this.handleClick}
              className={
                this.props.sizes.includes("XXL")
                  ? "size-filter size-filter-select"
                  : "size-filter"
              }
              value="XXL"
            />
          </div>
        </div>
      </div>
    );
  }
}

export default SizeFilters;
