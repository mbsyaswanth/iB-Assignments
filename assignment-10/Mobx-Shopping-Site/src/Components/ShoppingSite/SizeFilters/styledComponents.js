import styled from "styled-components";

export const SizeFiltersContainer = styled.div`
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  padding-left: 30px;
  flex-direction: column;
  align-items: flex-end;
`;

export const SizeFilter = styled.input`
  padding: 5px;
  margin: 5px;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  text-align: center;
  border-style: none;
  transition: 0.3s;
  cursor: pointer;
  :focus {
    outline: none;
  }
  :hover {
    box-shadow: 0px 0px 1px 2px black;
  }
  :active {
    border-style: none;
  }
  color: ${props => (props.select ? "white" : "black")};
  background-color: ${props =>
    props.select ? "black" : "rgba(219, 219, 219, 0.384)"};
`;
