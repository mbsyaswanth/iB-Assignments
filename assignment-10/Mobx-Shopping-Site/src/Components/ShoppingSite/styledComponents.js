import styled from "styled-components";

export const ShoppingSiteContainer = styled.div`
  padding-top: 50px;
  display: flex;
`;

export const FiltersSidebar = styled.div`
  width: 15%;
  padding-right: 5px;
`;

export const SizesHeading = styled.h3`
  padding-left: 40px;
`;

export const MainContentContainer = styled.div`
  width: 85%;
`;

export const SortBy = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 10px;
`;
