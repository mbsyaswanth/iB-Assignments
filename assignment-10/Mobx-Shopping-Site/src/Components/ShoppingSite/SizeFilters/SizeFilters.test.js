import React from "react";
import { shallow, mount, render } from "enzyme";
import "jest-styled-components";
import Products from "../../../Store";
import { sizeConstants } from "../../../constants";
import SizeFilters from "./index";
import { SizeFilter } from "./styledComponents";

const store = new Products();

describe("sizeFilters", () => {
  it("should Test whether add to cart function is called upon click", () => {
    const storeFunction = jest.spyOn(store, "updateSizeFilters");
    const component = mount(
      <SizeFilters
        onFilterClick={store.updateSizeFilters}
        sizes={store.sizes}
      />
    );
    //console.log(component.find(SizeFilter).debug());
    component
      .find(SizeFilter)
      .at(0)
      .simulate("click");
    expect(storeFunction).toHaveBeenCalled();
    console.log(store.sizes);

    component.unmount();
  });

  it("should change style on click - 1", () => {
    const component = mount(
      <SizeFilters
        onFilterClick={store.updateSizeFilters}
        sizes={store.sizes}
      />
    );
    expect(component.find(SizeFilter).at(0)).toHaveStyleRule(
      "background-color",
      "black"
    );
    component.unmount();
  });

  it("should change style on click - 2", () => {
    const component = mount(
      <SizeFilters
        onFilterClick={store.updateSizeFilters}
        sizes={store.sizes}
      />
    );
    component
      .find(SizeFilter)
      .at(0)
      .simulate("click");
    expect(component.find(SizeFilter).at(0)).toHaveStyleRule(
      "background-color",
      "rgba(219,219,219,0.384)"
    );
    component.unmount();
  });

  it("should have the same snapshot", () => {
    const component = shallow(
      <SizeFilters
        onFilterClick={store.updateSizeFilters}
        sizes={store.sizes}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
