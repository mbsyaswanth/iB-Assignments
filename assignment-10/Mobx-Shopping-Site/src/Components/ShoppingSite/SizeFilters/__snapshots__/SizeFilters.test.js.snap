// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sizeFilters should have the same snapshot 1`] = `
<styled.div>
  <div>
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="XS"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="S"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="M"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="ML"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="L"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="XL"
    />
    <styled.input
      onClick={[Function]}
      select={false}
      type="button"
      value="XXL"
    />
  </div>
</styled.div>
`;
